apis/management.cattle.io/v3/backup_types.go

```diff
@@ -0,0 +1,68 @@
+package v3
+
+import (
+	"github.com/rancher/norman/condition"
+	"github.com/rancher/norman/types"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	BackupConditionCreated   condition.Cond = "Created"
+	BackupConditionCompleted condition.Cond = "Completed"
+	BackupConditionRestored  condition.Cond = "Restored"
+)
+
+type BackupConfig struct {
+	// Backup interval in hours
+	IntervalHours int `yaml:"interval_hours" json:"intervalHours,omitempty" norman:"default=12"`
+	// Number of backups to keep
+	Retention int `yaml:"retention" json:"retention,omitempty" norman:"default=6"`
+	// s3 target
+	S3BackupConfig *S3BackupConfig `yaml:",omitempty" json:"s3BackupConfig"`
+}
+
+type S3BackupConfig struct {
+	// Access key ID
+	AccessKey string `yaml:"access_key" json:"accessKey,omitempty"`
+	// Secret access key
+	SecretKey string `yaml:"secret_key" json:"secretKey,omitempty" norman:"required,type=password" `
+	// name of the bucket to use for backup
+	BucketName string `yaml:"bucket_name" json:"bucketName,omitempty"`
+	// AWS Region, AWS spcific
+	Region string `yaml:"region" json:"region,omitempty"`
+	// Endpoint is used if this is not an AWS API
+	Endpoint string `yaml:"endpoint" json:"endpoint"`
+}
+type EtcdBackup struct {
+	types.Namespaced
```

Comment:
we need to add a clusterid as a field

Reply:
This field is so the UI can filter out the etcd backups on per cluster. And on the rancher api side, we should error out if we try to recover to the backups from other clusters.